components/brave_ads/browser/ads_service.h

```diff
@@ -31,6 +51,32 @@ class AdsService : public KeyedService {
   virtual uint64_t GetAdsPerHour() const = 0;
   virtual void SetAdsPerHour(const uint64_t ads_per_hour) = 0;
 
+  virtual void GetAdsHistory(OnGetAdsHistoryCallback callback) = 0;
+
+  virtual void ToggleAdThumbUp(const std::string& id,
```

Comment:
Rather than ToggleAdThumbUp, ToggleAdThumbDown etc. could we not just have something like OnAction(id, creative_set_id, action, type) where type is the type of action, this way we can easily add further types in the future

Reply:
Right, will have to think about how to approach that... The main problem is that the various actions take different parameters (for example, the opt in action expects a category name). I suppose I could have a map of parameters.